server/src/main/java/com/distkv/server/storeserver/runtime/expire/ExpirationManager.java

```diff
@@ -0,0 +1,142 @@
+package com.distkv.server.storeserver.runtime.expire;
+
+import com.distkv.common.exception.DistkvException;
+import com.distkv.rpc.protobuf.generated.DistkvProtocol.DistkvRequest;
+import com.distkv.rpc.protobuf.generated.DistkvProtocol.RequestType;
+import com.distkv.rpc.protobuf.generated.ExpireProtocol.ExpireRequest;
+import com.distkv.server.storeserver.StoreConfig;
+import com.google.protobuf.InvalidProtocolBufferException;
+import java.util.PriorityQueue;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ExpirationManager {
+
+  private static Logger LOGGER = LoggerFactory.getLogger(ExpirationManager.class);
+
+  private static final Integer DEFAULT_CAPACITY = 2048;
+
+  //Scheduled cleaning tasks.
+  ScheduledExecutorService scheduledExecutor = Executors.newSingleThreadScheduledExecutor();
+
+  private StoreConfig storeConfig;
+
+  /**
+   * PriorityQueue allows the data with the lowest expiration time to be queued. Just look at the
+   * cached most recent expired data and avoid scanning all caches.
+   */
+  public PriorityQueue<Node> expirationQueue = new PriorityQueue<>(DEFAULT_CAPACITY);
+
+  public ExpirationManager() {
+
+    /*
+     * Use the default scheduled executor to clear outdated data every 1 seconds.
+     */
+    scheduledExecutor.scheduleAtFixedRate(new SwapExpiredNode(), 0, 1, TimeUnit.SECONDS);
+  }
+
+  /**
+   * The expire request will be stored in the queue as a Node object. This internal Node object
+   * contains the key, expiration time and request type.
+   *
+   * @param request A expire request.
+   * @param storeConfig Local server config information.
+   */
+  public void addToCycle(DistkvRequest request, StoreConfig storeConfig) {
```

Comment:
Why pass storeConfig in this method?

I think storeConfig should be passed in constructor.